pilot/pkg/model/push_context.go

```diff
@@ -763,6 +770,18 @@ func (ps *PushContext) initServiceAccounts(env *Environment, services []*Service
 	}
 }
 
+// Caches list of authentication policies
+func (ps *PushContext) initAuthPolicies(env *Environment, services []*Service) {
+	for _, svc := range services {
+		if ps.AuthNPolicies[svc.Hostname] == nil {
+			ps.AuthNPolicies[svc.Hostname] = map[int]*Config{}
+		}
+		for _, port := range svc.Ports {
+			ps.AuthNPolicies[svc.Hostname][port.Port] = env.IstioConfigStore.AuthenticationPolicyForWorkload(svc, nil, port)
```

Comment:
nit: can you do one single List call in the beginning and then add the logic to determine authN be service/port as part of the loop? like how sidecar, envoyfilter, etc do